scene/3d/collision_object_3d.cpp

```diff
@@ -110,6 +111,41 @@ void CollisionObject3D::_update_pickable() {
 	}
 }
 
+void CollisionObject3D::_update_debug_shapes() {
+	for (Set<uint32_t>::Element *shapedata_idx = debug_shapes_to_update.front(); shapedata_idx; shapedata_idx = shapedata_idx->next()) {
+		if (shapes.has(shapedata_idx->get())) {
+			ShapeData shapedata = shapes[shapedata_idx->get()];
```

Comment:
This should be `ShapeData &shapedata`, right now this code is modifying a copy of the shape data.